### ISSUE TITLE
Feature Request: Schema stitching

### ISSUE THREAD

**Issue by jameskleeh** [graphql-java/graphql-java#952]
See the article here for the details of the feature:

https://dev-blog.apollodata.com/graphql-schema-stitching-8af23354ac37

**Comment by bbakerman** [comment]
Yup its a super hot topic.  I am not sure it belongs in the base graphql-java engine library but rather in a higher level library that uses graphql-java.

Watch this space.  I know of 2 efforts in Atlassian that are looking closely at this in Java and who knows what else might come from Apollo

**Comment by jiayou50** [comment]
@bbakerman I'm investigating on implementing schema stitching with graphql-java, and one major thing that seems to be missing is the ability to build a `GraphQLSchema` object from an introspection query result which is in JSON. This would correspond to the `buildClientSchema` function in the reference `graphql-js` implementation. 

I saw the `SchemaParser` class, but it only parses schemas written in the GraphQL schema language. Am I missing anything here?

**Comment by bbakerman** [comment]
I currently don't have access to a computer as I am on vacation...but I
think you are missing something... I recall a JSON to to schema
thing...perhaps it was printing only

The trick is how do you offer data fetching behaviour from introspection?

**Comment by mrdon-atlassian** [comment]
We (Atlassian) built a schema stitching library called graphql-braid: https://bitbucket.org/atlassian/graphql-braid/src

**Comment by benneq** [comment]
@mrdon-atlassian Is there any working example for your library?
The examples on your git README.md rely on non-existing classes ( `HttpGraphQLRemoteRetriever ` )
And use lot's of code that's not compatible with version `0.11.0` , `0.11.1`, `0.11.2`, `0.12.0`.

The `GraphQLRemoteSchemaSource` need some kind of `Supplier<Reader> schemaProvider`. In the examples of your `README.md` there's nothing like that.

I couldn't yet figure out how to use this... :(

**Comment by coolbeevip** [comment]
> We (Atlassian) built a schema stitching library called graphql-braid: https://bitbucket.org/atlassian/graphql-braid/src
hi, Is this project alive?

**Comment by pritamroy11** [comment]
is there any plan for implement schema stitching in graphql-java?

**Comment by bbakerman** [comment]
> is there any plan for implement schema stitching in graphql-java?

In graphql-java itself there is NO plan to support stitching.  There is a JVM based project for Apollo Federation spec stitching

https://github.com/apollographql/federation-jvm

We at Atlassian have released our Nadel stitching model via https://github.com/atlassian-labs/nadel

The thing about our Nadel library is that it is in very active development and does not have the API stability one might expect from an open source lib (since our goal is to build out as fast as we can for our own purposes and not to make it a stable in an OSS project sense)
